admin/workspace-providers/deployment.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Deployment
+description: Learn how to deploy a workspace provider.
+state: beta
+---
+
+This article walks you through the process of deploying a workspace provider to
+a [Kubernetes cluster](../../setup/kubernetes/index.md).
+
+[Workspace Providers](index.md) are logical groups of resources to which
+developers can deploy workspaces. Like the Coder deployment, workspace providers
+are deployed via a helm chart to the Kubernetes cluster where you'd like to
+provision new workspaces.
+
+## Dependencies
+
+Install the following dependencies if you haven't already:
+
+- [Coder CLI](../../cli/installation.md)
+- [helm](https://helm.sh/docs/intro/install/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Requirements
+
+1. Workspace providers **must have a hostname set** that is a subdomain of the
+   Coder deployment. For example, if the Coder deployment's hostname is
+   `coder.example.com`, the workspace provider's hostname must match the format
+   `*.coder.example.com`.
+2. The main Coder deployment and the workspace provider must be able to
+   communicate bi-directionally via their respective hostnames.
+3. The Kubernetes cluster address must be reachable from the Coder deployment.
+
+## Connecting to the Cluster
+
+To add a Kubernetes cluster as a workspace provider, you must first make sure
+that you're connected to the cluster you want to expand into
+
+```bash
+kubectl config current-context
+```
+
+Confirm that your current kubectl context correct before continuing; otherwise,
+connect to the correct context.
+
+## Creating the Coder Namespace (Optional)
+
+We recommend running workspace providers in a separate
+[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/);
+to do so, run
+
+```bash
+kubectl create namespace coder
+```
+
+Next, change the kubectl context to point to your newly created namespace:
+
+```bash
+kubectl config set-context --current --namespace=coder
+```
+
+## Creating the New Workspace Provider
+
+Using the Coder CLI, create a new workspace provider in the `pending` state.
+
+```bash
+coder providers create [NAME] \
+    --hostname=[HOSTNAME] \
+    --clusterAddress=[CLUSTER_ADDRESS]
```

Comment:
```suggestion
    --cluster-address=[CLUSTER_ADDRESS]
```

The argument is hyphenated as-of https://github.com/cdr/coder-cli/releases/tag/v1.17.0-rc.1